vortex-mask/src/lib.rs

```diff
@@ -509,6 +509,36 @@ impl Mask {
             _ => None,
         }
     }
+
+    /// Limit the mask to the first `limit` true values
+    pub fn limit(self, limit: usize) -> Self {
+        if self.len() <= limit {
+            return self;
+        }
+
+        match self {
+            Mask::AllTrue(len) => {
+                Self::from_iter([Self::new_true(limit), Self::new_false(len - limit)])
+            }
+            Mask::AllFalse(_) => self,
+            Mask::Values(ref mask_values) => {
+                if limit >= mask_values.true_count() {
+                    return self;
+                }
+
+                let existing_buffer = mask_values.boolean_buffer();
+
+                let mut new_buffer_builder = BooleanBufferBuilder::new(mask_values.len());
+                new_buffer_builder.append_n(existing_buffer.len(), false);
```

Comment:
nit: use mask_values.len() here, it's confusing that existing buffer is used here to mean the same thing as mask_values line above